### ISSUE TITLE
speaker verification with GE2E loss

### ISSUE THREAD

**Issue by wrongbattery** [mravanelli/SincNet#54]
I use your Sincnet architecture with GE2E loss but loss value can't drop. I try to replace activation unit in DNN to logistic function but it doesn't work?
Can you give me some advice why this happen? or just because that loss function can't cooperate with sincnet ?

**Comment by mravanelli** [comment]
Hi,
have tried to feed the network with fbanks instead of raw features? It could be important to know if the issue is due to the SincNet layer or something else. I haven't implemented GE2E loss, but I don't see any incompatibility with SincNet

**Comment by wrongbattery** [comment]
I tried what you suggested. Currently, filterbank + LSTM+ GE2E works, filterbank + cnn+ softmax works, but filterbank + cnn+ GE2E not works

**Comment by mravanelli** [comment]
Maybe there is something wrong in the CNN part or you simply have to tune
the hyperparameter (e.g., the learning rate) better.

**Comment by wrongbattery** [comment]
Have you try your sincnet with input of audio + noise? In all experiment, i use audio + musan noise. i try filterbank + your DNN code + GE2E => works, filterbank + your cnn code without sincnet_fast+ GE2E => loss decrease a little then soon diverge.

**Comment by mravanelli** [comment]
You aren't using SincNet here, right? In your fbank+cnn experiment it is
likely you just  have to tune a bit better the hyperparameters (maybe try
to reduce the learning rate). In general SincNet works well with noisy
inputs as well.

**Comment by wrongbattery** [comment]
I tried raw + sincnet + GE2E loss for the first time, but it doesn't work. later, you suggested me to test on fbanks. If filterbank + cnn +  GE2E doesn't, it's high chance that SincNet can't work with GE2E loss. I have tried to tune hyperparameters + model architecture to make it works. i will update the result soon

**Comment by mravanelli** [comment]
Hi, any updates?

**Comment by wrongbattery** [comment]
Hi @mravanelli,
It seems like this loss isn't stable enough like softmax for cnn to work. In GE2E, a sample need to move closer to its centroid and far from other centroids and theses centroids aren't fixed, there is no clue to know if class 1 in batch 1 is same as class 2 in batch... However, with softmax, these centroids are fixed

**Comment by wrongbattery** [comment]
with PASE model, i think i can modify Sincnet to make it work with GE2E loss. However, i'm quite busy at work so it will take time to get some updated results
